### PR TITLE
Add nfc example.

### PR DESCRIPTION
**Pull request by averne**
Requires https://github.com/switchbrew/libnx/pull/237.

### CODE REVIEW

File: nfc/source/amiibo_char.h [hunk 1]
```diff
@@ -0,0 +1,567 @@
+#ifndef _AMIIBO_CHAR_H
```
Comment (yellows8) [inline nfc/source/amiibo_char.h]: Remove this file.

File: nfc/source/main.c [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    }
+    
+    if (R_SUCCEEDED(rc)) {
+        PRINT_UPD("You can now remove the tag.\n");
+        eventWaitLoop(&deactivate_event);
+    }
+
+    // If an error happened during detection/reading, print it.
+    if (R_FAILED(rc))
+        PRINT_UPD("Error: 0x%x (%04d-%04d).\n", rc, (R_MODULE(rc) | 2000), (R_VALUE(rc) >> 9));
```
Comment (yellows8) [inline nfc/source/main.c]: Just print the raw rc.

File: nfc/source/amiibo_char.h (hunk 1 above)
Comment (averne) [reply nfc/source/amiibo_char.h]: Ok, was worried about possible copyright issues.

Comment (averne) [reply nfc/source/amiibo_char.h]: Wait I misread...

File: nfc/source/main.c [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+#include <stdlib.h>
+#include <string.h>
+#include <malloc.h>
+
+// Include the main libnx system header, for Switch development
+#include <switch.h>
+
+// See also libnx nfc.h.
+
+#define PRINT_UPD(fmt, ...) ({   \
```
Comment (yellows8) [inline nfc/source/main.c]: remove

**Comment by yellows8** [comment]
Shouldn't this be using nfpuGetDeviceState somewhere?

**Comment by yellows8** [comment]
Is merging the code using eventWaitLoop into a single main-loop an option? See also: https://github.com/devkitPro/3ds-examples/blob/master/nfc/source/main.c

**Comment by averne** [comment]
Sure. The idea was to keep this as simple as possible (hence why I didn't check nfpuGetState and nfpuDeviceState also).

**Comment by yellows8** [comment]
Also add appdata reading.

**Comment by yellows8** [comment]
Reminder that this still needs finished.

File: nfc/source/main.c [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+#include <malloc.h>
+
+// Include the main libnx system header, for Switch development
+#include <switch.h>
+
+// See also libnx nfc.h.
+
+// Hardcoded for Super Smash Bros. Ultimate.
+// See also https://switchbrew.org/wiki/NFC_services#Application_IDs.
+#define APP_ID 0x34f80200
```
Comment (yellows8) [inline nfc/source/main.c]: local var

File: nfc/source/main.c [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+        if (R_SUCCEEDED(rc) || (hidKeysDown(CONTROLLER_P1_AUTO) & KEY_PLUS))
+            break;
+    }
+    return rc;
+}
+
+// Print raw data as hexadecimal numbers.
+void print_hex(void *buf, size_t size) {
+    for (size_t i=0; i<size; i++)
+        printf("%02X", ((char *)buf)[i]);
```
Comment (yellows8) [inline nfc/source/main.c]: (u8*)

File: nfc/source/main.c [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+    // Get the deactivation event. This is signaled when a tag is removed.
+    Event deactivate_event = {0};
+    if (R_FAILED(nfpuAttachDeactivateEvent(controller, &deactivate_event)))
+        goto fail_1;
+
+    NfpuState state = 0;
+    if (R_SUCCEEDED(rc)) {
+        rc = nfpuGetState(&state);
+
+        if (R_SUCCEEDED(rc) && (state == NfpuState_NonInitialized)) {
```
Comment (yellows8) [inline nfc/source/main.c]: Remove () around state check.

File: nfc/source/main.c [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+            consoleUpdate(NULL);
+            rc = -1;
+        }
+    }
+
+    NfpuDeviceState device_state = 0;
+    if (R_SUCCEEDED(rc)) {
+        rc = nfpuGetDeviceState(controller, &device_state);
+
+        if (R_SUCCEEDED(rc) && (device_state > NfpuDeviceState_TagFound)) {
```
Comment (yellows8) [inline nfc/source/main.c]: See above.

File: nfc/source/main.c [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+// Indefinitely wait for an event to be signaled
+// Break when + is pressed, or if the application should quit (in this case, return value will be non-zero)
+Result eventWaitLoop(Event *event) {
+    Result rc = 1;
+    while (appletMainLoop()) {
+        rc = eventWait(event, 0);
+        hidScanInput();
+        if (R_SUCCEEDED(rc) || (hidKeysDown(CONTROLLER_P1_AUTO) & KEY_PLUS))
+            break;
+    }
```
Comment (yellows8) [inline nfc/source/main.c]: consoleUpdate()

File: nfc/source/main.c [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+            break;
+        consoleUpdate(NULL);
+    }
+    return rc;
+}
+
+// Print raw data as hexadecimal numbers.
+void print_hex(void *buf, size_t size) {
+    for (size_t i=0; i<size; i++)
+        printf("%02X", ((u8 *)buf)[i]);
```
Comment (yellows8) [inline nfc/source/main.c]: Remove space after u8.

File: nfc/source/main.c [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+    return rc;
+}
+
+// Main program entrypoint
+int main(int argc, char* argv[])
+{
+    Result rc = 0;
+
+    // Hardcoded for Super Smash Bros. Ultimate.
+    // See also https://switchbrew.org/wiki/NFC_services#Application_IDs.
```
Comment (yellows8) [inline nfc/source/main.c]: "See also: https://switchbrew.org/wiki/NFC_services#Application_IDs"

**Pull request merged**